util/codec/codec_test.go

```diff
@@ -520,54 +512,57 @@ func (s *testCodecSuite) TestBytes(c *C) {
 		{[]byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x00}, []byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08}, 1},
 	}
 
-	for _, t := range tblCmp {
-		b1 := EncodeBytes(nil, t.Arg1)
-		b2 := EncodeBytes(nil, t.Arg2)
+	for _, bytesData := range tblCmp {
+		b1 := EncodeBytes(nil, bytesData.Arg1)
+		b2 := EncodeBytes(nil, bytesData.Arg2)
 
 		ret := bytes.Compare(b1, b2)
-		c.Assert(ret, Equals, t.Ret)
+		require.Equal(t, bytesData.Ret, ret)
 
-		b1 = EncodeBytesDesc(nil, t.Arg1)
-		b2 = EncodeBytesDesc(nil, t.Arg2)
+		b1 = EncodeBytesDesc(nil, bytesData.Arg1)
+		b2 = EncodeBytesDesc(nil, bytesData.Arg2)
 
 		ret = bytes.Compare(b1, b2)
-		c.Assert(ret, Equals, -t.Ret)
+		require.Equal(t, -bytesData.Ret, ret)
 	}
 }
 
-func parseTime(c *C, s string) types.Time {
+func parseTime(t *testing.T, s string) types.Time {
 	sc := &stmtctx.StatementContext{TimeZone: time.UTC}
 	m, err := types.ParseTime(sc, s, mysql.TypeDatetime, types.DefaultFsp)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	return m
 }
 
-func parseDuration(c *C, s string) types.Duration {
+func parseDuration(t *testing.T, s string) types.Duration {
 	m, err := types.ParseDuration(nil, s, types.DefaultFsp)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	return m
 }
 
-func (s *testCodecSuite) TestTime(c *C) {
-	defer testleak.AfterTest(c)()
+func TestTime(t *testing.T) {
+	t.Parallel()
+
 	tbl := []string{
 		"2011-01-01 00:00:00",
 		"2011-01-01 00:00:00",
 		"0001-01-01 00:00:00",
 	}
 	sc := &stmtctx.StatementContext{TimeZone: time.Local}
-	for _, t := range tbl {
-		m := types.NewDatum(parseTime(c, t))
+	for _, timeDatum := range tbl {
+		m := types.NewDatum(parseTime(t, timeDatum))
 
 		b, err := EncodeKey(sc, nil, m)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 		v, err := Decode(b, 1)
-		c.Assert(err, IsNil)
-		var t types.Time
-		t.SetType(mysql.TypeDatetime)
-		err = t.FromPackedUint(v[0].GetUint64())
-		c.Assert(err, IsNil)
-		c.Assert(types.NewDatum(t), DeepEquals, m)
+		require.NoError(t, err)
+
+		var rawTime types.Time
+		rawTime.SetType(mysql.TypeDatetime)
+		err = rawTime.FromPackedUint(v[0].GetUint64())
+		require.NoError(t, err)
+
+		require.Equal(t, types.NewDatum(rawTime), m)
```

Comment:
I think the expected is `m`, and it should be `Equal(t, m, types.NewDatum(rawTime))`?

Reply:
correct. thanks, already updated.